src/V3Timing.cpp

```diff
@@ -345,19 +388,33 @@ class TimingSuspendableVisitor final : public VNVisitor {
             if (depVxp->nodep()->user2() & T_SUSPENDEE) propagateFlags(depVxp, T_SUSPENDEE);
         }
         if (dumpGraphLevel() >= 6) m_suspGraph.dumpDotFilePrefixed("timing_deps");
-        // Propagate process
+
+        // Propagate needProcess
         for (V3GraphVertex* vxp = m_procGraph.verticesBeginp(); vxp; vxp = vxp->verticesNextp()) {
             DepVtx* const depVxp = static_cast<DepVtx*>(vxp);
-            if (depVxp->nodep()->user2() & T_HAS_PROC) propagateFlags(depVxp, T_HAS_PROC);
+            // Find processes that'll allocate VlProcess
+            if ((depVxp->nodep()->user2() & T_FORCES_PROC)) {
+                propagateFlagsIf(depVxp, T_FORCES_PROC, [&](const V3GraphEdge* e) -> bool {
+                    return !(static_cast<DepVtx*>(e->fromp())->nodep()->user2() & T_ALLOCS_PROC);
```

Comment:
Just a thought, but since we have flag setters, perhaps flag getters would make the code a bit more readable? It would be more consistent and the abstraction would be less leaky. Not sure tho.

Reply:
I did it. But the scope of that is limited, {add,has}Flags belong to the graph-building object.

Reply:
Should they belong to it? The enum doesn't. The functions are static.

Reply:
Good point. I took them out of the class.